rest/route/host_create.go

```diff
@@ -70,7 +70,40 @@ func (h *hostCreateHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	ids := []string{}
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	initialHosts, err := host.FindHostsSpawnedByTask(ctx, h.taskID, t.Execution)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding hosts spawned by task '%s'", h.taskID))
+	}
+	numInitialHosts := len(initialHosts)
 	for i := 0; i < numHosts; i++ {
+		// If there are an unexpected number of host intents, log a message and continue.
+		spawnedHosts, err := host.FindHostsSpawnedByTask(ctx, h.taskID, t.Execution)
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding hosts spawned by task '%s'", h.taskID))
+		}
+
+		if len(spawnedHosts) != numInitialHosts+i {
```

Comment:
This check seems a little odd because what we really want is to have exactly `numHosts` hosts and not any more/less, so even if one REST request partially succeeded, a retry should allow it to make the remaining hosts it needs without issue. For example, if the first REST request managed to make 2 hosts, and then the next REST request made 3, that shouldn't log. Would it make more sense to check that `len(spawnedHosts) < numHosts`?